storage/block.go

```diff
@@ -52,6 +54,10 @@ type dbBlock struct {
 
 // NewDatabaseBlock creates a new DatabaseBlock instance.
 func NewDatabaseBlock(start time.Time, encoder m3db.Encoder, opts m3db.DatabaseOptions) m3db.DatabaseBlock {
+	srAlloc := xio.NewSegmentReader
+	if pool := opts.GetSegmentReaderPool(); pool != nil {
```

Comment:
To save a pointer here why don't we just assume we always have a segment reader pool?  We're not in the encoding package so I think it's ok to make assumptions.  Also we do this a lot of other places too inside the storage package, assuming there is a pool related that is.  The defaults which include creating a segment reader pool in `storage/options.go` are local to the package so we're not making an assumption outside of the package too which is good.

I'm a fan of just initializing the pools to size zero and still using them rather than nil checking them everywhere.